modules/standard/IO.chpl

```diff
@@ -7583,13 +7618,19 @@ proc fileReader.readThrough(separator: string, ref s: string, maxSize=-1, stripS
   on this._home {
     try this.lock(); defer { this.unlock(); }
 
+    if separator.numBytes == 0 {
+      throw new IllegalArgumentError("readThrough(string) called with empty separator");
+    }
+
     // performance TODO: investigate using qio_channel_read_string as a fast path for single-byte separators
     //  (this would be a single pass and would not require retroactive codepoint checking)
 
     // find the byte offset to the start of the separator, 'maxSize' bytes, or EOF (whichever comes first)
     const (searchErr, found, bytesOffset) = _findSeparator(separator, 4*maxSize, this._channel_internal);
-    // handle system error
-    if searchErr != 0 && searchErr != EEOF then try this._ch_ioerror(searchErr, "in readThrough(string)");
+    // handle system error & not found within maxSize
```

Comment:
I might rename `bytesOffset` to `bytesRead` for consistency with your other changes (and because it's a more accurate name).